sections/loans/components/ActionBox/ActiveBorrowsTab/ModifyLoanActions/Deposit.tsx

```diff
@@ -54,94 +52,77 @@ const Deposit: React.FC<DepositProps> = ({
 		!amount ? setDepositalAmount(null) : setDepositalAmount(amount);
 	const onSetLeftColMaxAmount = (amount: string) => setDepositalAmount(amount);
 
-	const getApproveTxData = useCallback(() => {
-		if (!(collateralAssetContract && !depositAmount.eq(0))) return null;
-		return [collateralAssetContract, 'approve', [loanContractAddress, depositAmount.toBN()]];
-	}, [collateralAssetContract, loanContractAddress, depositAmount]);
-
-	const getDepositTxData = useCallback(() => {
-		if (!(loanContract && !depositAmount.eq(0))) return null;
-		return [
-			loanContract,
-			'deposit',
-			[walletAddress, loanId, ...(loanTypeIsETH ? [] : [depositAmount.toBN()])],
-			{ value: loanTypeIsETH ? depositAmount.toBN() : 0 },
-		];
-	}, [loanContract, walletAddress, loanId, loanTypeIsETH, depositAmount]);
-
-	const getTxData = useMemo(
-		() => (!isApproved ? getApproveTxData : getDepositTxData),
-		[isApproved, getApproveTxData, getDepositTxData]
+	const isApprovedQuery = useQuery(
+		'isApproved',
+		async () => {
+			const allowance = await collateralAssetContract.allowance(walletAddress, loanContractAddress);
+			return wei(allowance).gte(depositAmount);
+		},
+		{ enabled: Boolean(collateralAssetContract && loanContractAddress && walletAddress) }
+	);
+	const isApproved = loanTypeIsETH ? true : isApprovedQuery.data;
```

Comment:
loadTypeIsETH || isApprovedQuery.data could also work here.

Reply:
This logic will be removed in #1195 so not doing anything related to this